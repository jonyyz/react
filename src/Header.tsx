import * as React from "react";
import { Link } from "react-router-dom";

import {
  createStyles,
  withStyles
} from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import PersonIcon from "@material-ui/icons/Person";

const styles = createStyles({
  root: {
    flexGrow: 1
  },
  toolBar: {
    paddingLeft: 10,
    paddingRight: 10,
  },
  titleContainer: {
    textAlign: "left",
    flexGrow: 1,
    textTransform: "none"
  },
  personIcon: {
    marginRight: 3,
  }
});

interface IProps {
  classes?: any,
  title: string
}

const Header: React.FunctionComponent<IProps> = (props) => {
  const { classes } = props;

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar className={classes.toolBar}>
          <div className={classes.titleContainer}>
            <Typography variant="h5">{props.title}</Typography>
          </div>
          <Button
            type="submit"
            variant="contained"
            color="secondary"
            component={Link}
            to="/users"
          >
            <PersonIcon className={classes.personIcon} /> Get Users
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default withStyles(styles)(Header);
