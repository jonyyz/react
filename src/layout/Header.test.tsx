import * as React from "react";
import { shallow } from "enzyme";

import Header from "./Header";

describe("Header", () => {
  test("should render", () => {
    const wrapper = shallow(<Header title="Carbonite" />);
    expect(wrapper).toMatchSnapshot();
  });
});
