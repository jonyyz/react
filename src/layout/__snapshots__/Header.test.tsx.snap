// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toolBar-2"
    >
      <div
        className="makeStyles-titleContainer-3"
      >
        <WithStyles(ForwardRef(Typography))
          variant="h5"
        >
          Carbonite
        </WithStyles(ForwardRef(Typography))>
      </div>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        component={[Function]}
        to="/users"
        type="submit"
        variant="contained"
      >
        <PersonIcon
          className="makeStyles-personIcon-4"
        />
        Get Users
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
